components/script/dom/html/htmldetailselement.rs

```diff
@@ -285,6 +478,12 @@ impl VirtualMethods for HTMLDetailsElement {
                     }
                 }));
             self.upcast::<Node>().dirty(NodeDamage::Other);
+
+            if self.Open() {
```

Comment:
```suggestion
            // Step 3.2. If oldValue is null and value is not null,
            // then ensure details exclusivity by closing other elements if needed given element.
            if self.Open() {
```
Seems like we are not handling the case where `oldValue` is not null? We only check if `value` is null.